### ISSUE TITLE
Flag to only check AnnotatedFor() marked code

### ISSUE THREAD

**Issue by thisisalexandercook** [eisop/checker-framework#1272]
Consider the following classes:
```java
// file b.java
@AnnotatedFor("nullness")
class b {
    Object o = null;
}

```
```java
// file a.java
class a {
    Object o = null;
}
```

In the Checker Frameworks current setup, both classes will throw an  `[assignment.type.incompatible]` when compiled with the nullness checker. However, it would be useful to add a flag (e.g. `OnlyAnnotatedFor`) that instructs a checker to only check files that explicitly contain an `@AnnotatedFor()` marking. This behaviour would mirror Nullaways [OnlyNullMarked](https://github.com/uber/NullAway/wiki/Configuration) option and JSpecify's [Nullness Specification](https://jspecify.dev/docs/spec/). 

Additionally,`@AnnotatedFor("nullness")` should be aliased to JSpecify's `@NullMarked` annotation to provide support for projects using this mechanism.
